pymc3/step_methods/hmc/nuts.py

```diff
@@ -12,6 +12,7 @@
 
 
 def bern(p):
+    assert np.isfinite(p)
```

Comment:
That slipped in from debugging... But it might make sense to check for nan. infs should be fine.